### DISCUSSION TITLE
Path template configuration with regular expressions

### DISCUSSION THREAD

**Discussion by miqh** [ThreeMammals/Ocelot#1906] [Ideas]
Hello,

I went through the [documentation page on routing][1] and didn't notice any mention for using regular expressions in path template properties.

After haphazardly experimenting with different expressions for `UpstreamPathTemplate`, I noticed a simple expression like `(.+)` appeared to be able to match against input, but I couldn't figure out whether the captured group could be used in `DownstreamPathTemplate`. Or perhaps this was purely coincidental behaviour.

I just wanted to ask whether or not there was support for using regular expressions to capture fragments of a route. If there's any documentation I missed regarding this, please point me in the right direction.

Currently using version 12.0.1.

Thanks in advance.

[1]: https://ocelot.readthedocs.io/en/latest/features/routing.html

**Comment by philproctor** [comment]
This will be resolved as part of my work on https://github.com/ThreeMammals/Ocelot/issues/737

Currently regex will work in the path, but the capture groups cannot be sent to the downstream. After this work, this will be possible.

**Reply by raman-m** [reply]
How does it work? Please show an example and code!

**Reply by raman-m** [reply]
> This will be resolved as part of my work on https://github.com/ThreeMammals/Ocelot/issues/737

Aha! The #737 issue is closed by me. Reason: No communication.

**Comment by miqh** [comment]
@philproctor, that's great to hear! Thanks for the update. :raised_hands:

**Comment by GadgetGeekUK** [comment]
Would this allow routing restrictions, so that if certain values appear in the upstream route they will go to destination 1, if others appear go to destination 2?

Defining 2 ReRoutes is fine, but I want to have something like:

myurl/customer/(Bob|Andy)/accounts go to destination 1,
myurl/customer/(Other|Any|None)/accounts go to destination 2

Thanks.

**Comment by raman-m** [comment]
@miqh [on Jan 16, 2019](https://github.com/ThreeMammals/Ocelot/discussions/1906#discussion-6039921):
> I went through the [documentation page on routing](https://ocelot.readthedocs.io/en/latest/features/routing.html) and didn't notice any mention for using regular expressions in path template properties.

Regular expressions in templates are not supported/implemented!

-----
> After haphazardly experimenting with different expressions for `UpstreamPathTemplate`, I noticed a simple expression like `(.+)` appeared to be able to match against input, but I couldn't figure out whether the captured group could be used in `DownstreamPathTemplate`. Or perhaps this was purely coincidental behavior.

coincidental behavior

-----
>I just wanted to ask whether or not there was support for using regular expressions to capture fragments of a route. If there's any documentation I missed regarding this, please point me in the right direction.

No documentation at all! Not implemented!
If you need this feature &rarr; **Please contribute!**
